client.py

```diff
@@ -13,113 +15,192 @@
+                #### END TEXT INTERFACE ####
+
+            elif parsed_message.code == MessageType.SPIN:
+                [player_id] = parsed_message.args  # TODO: Why is this client receiving its own player ID?
+
+                # TODO (UI): Ask this player to push a button to spin the wheel
+                # TODO (UI): Delete text interface code below
+
+                ### BEGIN TEXT INTERFACE ###
+                print("Spin the wheel!")
+                input("Press enter to spin the wheel")
+                #### END TEXT INTERFACE ####
+
+                response_info = []
+
+            elif parsed_message.code == MessageType.END_GAME:
+                [winner_player_id] = parsed_message.args
+
+                # TODO (UI): Tell player the game has ended, and display the winning player id
+                # TODO (UI): Delete text interface code below
+
+                ### BEGIN TEXT INTERFACE ###
+                print("\n==========================\nEND OF GAME\n==========================\n")
+                print(f"Player {winner_player_id} has won!")
+                input("Press enter to end the game")
+                #### END TEXT INTERFACE ####
+
+                self.game_over = True
+                response_info = []
+
+            elif parsed_message.code == MessageType.UPDATE_SCORES:
+                [scores_dict, tokens_dict, num_spins] = parsed_message.args
+
+                # TODO (UI): Update the UI to display the new scores, number of tokens, and number of spins to players
+                # TODO (UI): Delete text interface code below
+
+                ### BEGIN TEXT INTERFACE ###
+                print("\nUPDATED GAME VALUES:")
+                print(f"Number of Spins: {num_spins}")
+                for player_id in scores_dict:
+                    print(f"Player {player_id}: {scores_dict[player_id]} points | {tokens_dict[player_id]} tokens")
+                print("")
+                #### END TEXT INTERFACE ####
+
+                response_info = []
+
+            else:
+                raise Exception(f"Client {self.player_id} received unknown MessageType: {parsed_message.code}")
+
+            self.send_command(server, Message(parsed_message.code, response_info))
+
+    def __prompt_user_from_list(self, prompt_list: list):
+        """
+        Print out the contents of a list, and prompt the user to pick one.
+        :param prompt_list: List of strings to pick from (e.g. a list of answers to a Jeopardy question
+        :return: (string) The answer that the user selected
+        """
+        # Print out the options in the list
+        for i in range(0, len(prompt_list)):
+            print("----- " + str(i) + ") " + str(prompt_list[i]))
+        # Ask user to select an option from the list
+        selected_index = None
+        while selected_index not in range(0, len(prompt_list)):
+            max_index = str(len(prompt_list))
```

Comment:
Since you're starting index from 0,
```suggestion
            max_index = str(len(prompt_list)-1)
```